test/requests/userAPI.spec.ts

```diff
@@ -0,0 +1,215 @@
+        describe('Request success', () => {
+            let user: User;
+
+            beforeEach(() => {
+                user = buildUser();
+            });
+
+            it('should returns message to user created with success', () => {
+                const email = user.email;
+                const password = user.password;
+
+                return request(expressApp)
+                .post('/api/user/signup')
+                .send({ email, password })
+                .expect('Content-Type', /json/)
+                .expect(200)
+                .then((response) => { 
+                    expect(response.body).to.deep.equal({ message: "user created with success" });
+                    totalUserByemail(email).then((total) => expect(total).to.equal(1));
+                });
+            });
+        });
+
+        describe('Validate request', () => {
+            describe('request send email user already exist', () => {
+                let user: User;
+
+                beforeEach(() => {
+                    return persistUser(buildUser()).then((userCreated) => {
+                        user = userCreated;
+                    }); 
+                });
+
+                it('should returns an error email already registered', () => {
+                    const email = user.email;
+                    const password = user.password;
+
+                    return request(expressApp)
+                    .post('/api/user/signup')
+                    .send({ email, password })
+                    .expect('Content-Type', /json/)
+                    .expect(400)
+                });
+            });
+
+            describe('request send email and password empty', () => {
+                it('should returns multiple errors if more than one field is required', () => {
+                    const email = '';
+                    const password = '';
+
+                    return request(expressApp)
+                    .post('/api/user/signup')
+                    .send({ email, password })
+                    .expect('Content-Type', /json/)
+                    .expect(400)
+                    .then((response) => expect(response.body).to.deep.equals(['email is required', 'password is required']))
+                });
+            });
+            describe('request send email and password invalid', () => {
+                it('should returns multiple errors if more than one field is invalid', () => {
+                    const email = 'paris';
+                    const password = '123ab';
+
+                    return request(expressApp)
+                    .post('/api/user/signup')
+                    .send({ email, password })
+                    .expect('Content-Type', /json/)
+                    .expect(400)
+                    .then((response) => expect(response.body).to.deep.equals(['email format is invalid', 'your password must be have at least: 8 characters long, 1 uppercase, 1 lowercase character and 1 number']))
```

Comment:
break this line; long lines are not good for readability. try to keep in the max 120chrs per line.